platforms/extensibility/plugin-use/src/main/java/org/gradle/plugin/use/resolve/internal/AlreadyOnClasspathPluginResolver.java

```diff
@@ -93,8 +95,14 @@ public void resolve(PluginRequestInternal pluginRequest, PluginResolutionResult
     }
 
     private void resolveAlreadyOnClasspath(PluginId pluginId, String pluginVersion, PluginResolutionResult result) {
-        PluginResolution pluginResolution = new ClassPathPluginResolution(pluginId, pluginVersion, parentLoaderScope, pluginInspector);
-        result.found("Already on classpath", pluginResolution);
+        DefaultPluginRegistry pluginRegistry = new DefaultPluginRegistry(pluginInspector, parentLoaderScope);
+        PluginImplementation<?> plugin = pluginRegistry.lookup(pluginId);
+        if (plugin != null) {
+            PluginResolution pluginResolution = new ClassPathPluginResolution(pluginId, pluginVersion, plugin);
+            result.found("Already on classpath", pluginResolution);
+        } else {
+            result.notFound("Classpath", "Plugin with id '" + pluginId + "' not found.");
```

Comment:
I'm not clear how this branch could even be hit.  The resolver seems to already found on the classpath or found an original request at this point.

Reply:
This check was already here in `ClassPathPluginResolution.java` but it was moved up to this class in order to use the proper `notFound` instead of throwing an exception.

Reply:
But shouldn't this cause an exception?  Why would there not be an plugin found here?

It seems to make sense in the original location `ClassPathPluginResolution`, but this is an **AlreadyOnClasspath** resolver - so shouldn't it already be in the registry?

Reply:
That is handled in `resolveToFoundResult`. We throw an exception if someone calls `notFound` and no other resolver calls `found`.

Reply:
An exception still ends up getting thrown, it just gets thrown in a separate place

Reply:
Lets add some javadoc to this class